<a href="https://colab.research.google.com/github/sananaz-git/city_of_whittlesea/blob/main/MyDataCleaning.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# City of Whittlesea Household survey Data Cleaning Task


Datasets:  

1.   Whittlesea - 2021 Household Survey - household_sav
2.   Whittlesea - 2021 Household Survey - person_sav


Survey Questions Cleaned by **Sana**:

1. Q3
2. Q9
3. Q10
4. Q15
5. Q18
6. Q21
7. Q22
8. Q23
9. Q33
10. Q37
11. Q40
12. Q42
13. Q45
14. Q50
15. Q54
16. Q27





In [ ]:
import pandas as pd
import numpy as np

In [ ]:
filepath = "/content/Whittlesea - 2021 Household Survey - person_sav.csv"
person_df = pd.read_csv(filepath)

In [ ]:
filepath = "/content/Whittlesea - 2021 Household Survey - household_sav.csv"
household_df = pd.read_csv(filepath)

In [ ]:
pd.options.display.max_rows = 1545
pd.options.display.max_columns = 400

In [ ]:
person_df.head() # First 5 records of person dataset

Q1 Selfdescribe Q13 Other Q18 Other Q24 Other               Q26      Q27  \
0             NaN       NaN       NaN       NaN  Aged care worker   Health   
1             NaN       NaN       NaN       NaN               NaN      NaN   
2             NaN       NaN       NaN       NaN       Taxi driver  Tourism   
3             NaN       NaN       NaN       NaN               NaN      NaN   
4             NaN       NaN       NaN       NaN      Truck Driver    Waste   

  Q3 Other Q30 Other Q34 Other Q35 Other Q36 Other  No of person  Person no  \
0      NaN       NaN       NaN       NaN       NaN             4          1   
1      NaN       NaN       NaN       NaN       NaN             4          2   
2      NaN       NaN       NaN       NaN       NaN             4          3   
3   Sister       NaN       NaN       NaN       NaN             4          4   
4      NaN       NaN       NaN       NaN       NaN             3          1   

   Precinct   Q1     Q10  Q11  Q12  Q13 1  Q13 10  Q13 11  Q13 12  Q13 2  \
0      11.0  2.0  3076.0  3.0  NaN      2       2       2       2      2   
1      11.0  2.0  3076.0  3.0  NaN      2       2       2       2      2   
2      11.0  1.0  3076.0  3.0  NaN      2       2       2       2      2   
3      11.0  1.0  3076.0  2.0  NaN      2       2       1       2      2   
4      10.0  1.0  3064.0  3.0  NaN      2       2       2       2      2   

   Q13 3  Q13 4  Q13 5  Q13 6  Q13 7  Q13 8  Q13 9  Q14  Q15  Q16  Q17  Q18 1  \
0      2      2      2      2      2      2      2  1.0  1.0  4.0  3.0      2   
1      2      2      2      2      2      2      2  NaN  5.0  6.0  3.0      2   
2      2      2      2      2      2      2      2  3.0  1.0  4.0  3.0      2   
3      2      2      2      2      2      2      2  1.0  5.0  3.0  3.0      2   
4      2      2      2      2      2      2      2  3.0  1.0  3.0  3.0      2   

   Q18 10  Q18 11  Q18 12  Q18 13  Q18 14  Q18 2  Q18 3  Q18 4  Q18 5  Q18 6  \
0       2       2       2       2       2      2      2      2      2      2   
1       2       2       2       2       2      2      2      2      2      2   
2       2       2       2       2       2      2      2      2      2      2   
3       2       2       2       2       2      2      2      2      2      2   
4       2       2       2       2       2      2      2      2      2      2   

   Q18 7  Q18 8  Q18 9  Q18 Disability  Q19 1  Q19 2  Q19 3  Q19 4  Q19 5  \
0      2      2      2               2      1      2      2      2      2   
1      2      2      2               2      1      2      2      2      2   
2      2      2      2               2      1      2      2      2      2   
3      2      2      2               2      1      2      2      2      2   
4      2      2      2               2      1      2      2      2      2   

   Q19 6  Q19 7  Q19 8    Q2  Q2 Cohorts  Q2 Lifecycle  Q20 1  Q20 2  Q20 3  \
0      2      2      2  35.0         8.0           5.0      2      1      2   
1      2      2      2   3.0         1.0           1.0      2      2      2   
2      2      2      2  36.0         8.0           5.0      2      2      2   
3      2      2      2  32.0         7.0           4.0      2      2      2   
4      2      2      2  32.0         7.0           4.0      2      2      2   

   Q20 4  Q20 5  Q20 6  Q20 7  Q20 8  Q20 9  Q21  Q22  Q23  Q24  Q25 1  Q25 2  \
0      2      2      1      2      2      2  4.0  NaN  2.0  2.0      1      2   
1      2      2      2      2      2      2  NaN  NaN  2.0  NaN      2      2   
2      2      2      2      2      2      2  5.0  NaN  2.0  2.0      1      2   
3      2      2      2      2      2      2  5.0  8.0  1.0  9.0      2      2   
4      2      2      2      2      2      2  3.0  NaN  2.0  1.0      1      2   

   Q25 3  Q25 4  Q25 5  Q25 6  Q25 7  Q25 9  Q28     Q29   Q3  Q30  Q31  Q32  \
0      2      2      2      2      2      2  5.0  3095.0  6.0  2.0  4.0  5.0   
1      2      2      2      2      2      2  NaN

In [ ]:
person_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1545 entries, 0 to 1544
Columns: 151 entries, Q1 Selfdescribe to Precinct.1
dtypes: float64(34), int64(104), object(13)
memory usage: 1.8+ MB


1545 rows and 151 columns in person dataset

In [ ]:
person_df.shape

(1545, 151)

#For Q3. What is the person's relationship to Person One?

In [ ]:
person_df.Q3.isna().sum() # 5 NAs

5

In [ ]:
# Creating separate categories for each option including one for NAs
person_df.Q3.value_counts()

6.0    606
3.0    458
1.0    404
4.0     41
2.0     20
5.0      8
9.0      3
Name: Q3, dtype: int64

From the survey questions the Personal Relationships frequency is as follows:


1. Self 
2. Child 
3. Partner 
4. Relative 
5. Parent 
6. Un-related Housemate 
7. Other 

Majority relationships are Children and Partners.

In [ ]:
# Convert Q3 from float to integers and create dummy variables including one category for missing values
person_df['Q3'] = pd.to_numeric(person_df['Q3'],errors='ignore').astype(pd.Int64Dtype())

In [ ]:
#Creating dummy variables for Q3:
dummy_vars = pd.get_dummies(person_df["Q3"],prefix="Q3",prefix_sep="_",dummy_na=True)
dummy_vars.head()

Q3_1  Q3_2  Q3_3  Q3_4  Q3_5  Q3_6  Q3_9  Q3_nan
0     0     0     0     0     0     1     0       0
1     0     0     1     0     0     0     0       0
2     1     0     0     0     0     0     0       0
3     0     0     0     1     0     0     0       0
4     0     0     0     0     0     1     0       0

Also verified if all relations from 'Q3 Other' got transferred in 'Q3_4' field, i.e. the option for 'Other relative'

In [ ]:
person_df.dropna(subset=['Q3 Other'])  # All relationships under 'Q3 Other' removing NANs

Q1 Selfdescribe Q13 Other Q18 Other Q24 Other          Q26          Q27  \
3                NaN       NaN       NaN       NaN          NaN          NaN   
836              NaN       NaN       NaN       NaN          NaN          NaN   
1003             NaN       NaN       NaN       NaN          NaN          NaN   
1058             NaN       NaN       NaN       NaN          NaN          NaN   
1101             NaN       NaN       NaN       NaN          NaN          NaN   
1209             NaN       NaN       NaN       NaN          NaN          NaN   
1219             NaN       NaN       NaN       NaN          NaN          NaN   
1371             NaN       NaN       NaN       NaN          NaN          NaN   
1372             NaN       NaN       NaN       NaN          NaN          NaN   
1373             NaN       NaN       NaN       NaN          NaN          NaN   
1535             NaN       NaN       NaN       NaN  Hospitality  Hospitality   
1536             NaN       NaN       NaN       NaN     Designer     Designer   

             Q3 Other Q30 Other Q34 Other Q35 Other Q36 Other  No of person  \
3              Sister       NaN       NaN       NaN       NaN             4   
836          Grandson       NaN       NaN       NaN       NaN             4   
1003       Grandchild       NaN       NaN       NaN       NaN             5   
1058           Sister       NaN       NaN       NaN       NaN             3   
1101       Grandchild       NaN       NaN       NaN       NaN             5   
1209         Grandson       NaN       NaN       NaN       NaN             5   
1219       Grandchild       NaN       NaN       NaN       NaN             3   
1371  Daughter in Law       NaN       NaN       NaN       NaN             6   
1372       Grandchild       NaN       NaN       NaN       NaN             6   
1373       Grandchild       NaN       NaN       NaN       NaN             6   
1535       Grandchild       NaN       NaN       NaN       NaN             4   
1536       Grandchild       NaN       NaN       NaN       NaN             4   

      Person no  Precinct   Q1     Q10  Q11  Q12  Q13 1  Q13 10  Q13 11  \
3             4      11.0  1.0  3076.0  2.0  NaN      2       2       1   
836           4       4.0  1.0  3076.0  3.0  NaN      2       2       2   
1003          5       3.0  1.0  3752.0  3.0  NaN      2       2       2   
1058          3       9.0  2.0     NaN  2.0  NaN      2       2       2   
1101          5       9.0  1.0     NaN  3.0  NaN      2       2       2   
1209          5       8.0  1.0  3082.0  3.0  NaN      2       2       2   
1219          3       8.0  1.0     NaN  2.0  NaN      2       2       2   
1371          4       5.0  2.0  3750.0  3.0  NaN      2       2       2   
1372          5       5.0  1.0  3750.0  3.0  NaN      2       2       2   
1373          6       5.0  2.0  3750.0  3.0  NaN      2       2       2   
1535          3       1.0  2.0     NaN  3.0  NaN      2       2       2   
1536          4       1.0  1.0     NaN  3.0  NaN      2       2       2   

      Q13 12  Q13 2  Q13 3  Q13 4  Q13 5  Q13 6  Q13 7  Q13 8  Q13 9  Q14  \
3          2      2      2      2      2      2      2      2      2  1.0   
836        2      2      2      2      2      2      2      2      2  3.0   
1003       2      2      2      2      2      2      2      2      2  3.0   
1058       1      2      2      2      2      2      2      2      2  3.0   
1101       2      2      2      2      2      2      2      2      2  3.0   
1209       2      2      2      2      2      2      2      2      2  3.0   
1219       2      2      2      2      2      1      2      2      2  3.0   
1371       2      2      2      2      2      2      2      2      2  1.0   
1372       2      2      2      2      2      2      2      2      2  3.0   
1373       2      2      2      2      2      2      2      2      2  3.0   
1535       2      1      2      2      2      2      2      2      2  3.0   
1536       2      1      2      2   

Dropping 'Q3' and 'Q3 Other' columns as it is original column and indicator variables have been created. Also Q3_6 is not required as it is respondents relationship to self.

In [ ]:
dummy_vars = dummy_vars.drop(["Q3_6"],axis = 1)

In [ ]:
person_df = person_df.drop(["Q3","Q3 Other"],axis = 1)
person_df = person_df.join(dummy_vars)
person_df.head()

Q1 Selfdescribe Q13 Other Q18 Other Q24 Other               Q26      Q27  \
0             NaN       NaN       NaN       NaN  Aged care worker   Health   
1             NaN       NaN       NaN       NaN               NaN      NaN   
2             NaN       NaN       NaN       NaN       Taxi driver  Tourism   
3             NaN       NaN       NaN       NaN               NaN      NaN   
4             NaN       NaN       NaN       NaN      Truck Driver    Waste   

  Q30 Other Q34 Other Q35 Other Q36 Other  No of person  Person no  Precinct  \
0       NaN       NaN       NaN       NaN             4          1      11.0   
1       NaN       NaN       NaN       NaN             4          2      11.0   
2       NaN       NaN       NaN       NaN             4          3      11.0   
3       NaN       NaN       NaN       NaN             4          4      11.0   
4       NaN       NaN       NaN       NaN             3          1      10.0   

    Q1     Q10  Q11  Q12  Q13 1  Q13 10  Q13 11  Q13 12  Q13 2  Q13 3  Q13 4  \
0  2.0  3076.0  3.0  NaN      2       2       2       2      2      2      2   
1  2.0  3076.0  3.0  NaN      2       2       2       2      2      2      2   
2  1.0  3076.0  3.0  NaN      2       2       2       2      2      2      2   
3  1.0  3076.0  2.0  NaN      2       2       1       2      2      2      2   
4  1.0  3064.0  3.0  NaN      2       2       2       2      2      2      2   

   Q13 5  Q13 6  Q13 7  Q13 8  Q13 9  Q14  Q15  Q16  Q17  Q18 1  Q18 10  \
0      2      2      2      2      2  1.0  1.0  4.0  3.0      2       2   
1      2      2      2      2      2  NaN  5.0  6.0  3.0      2       2   
2      2      2      2      2      2  3.0  1.0  4.0  3.0      2       2   
3      2      2      2      2      2  1.0  5.0  3.0  3.0      2       2   
4      2      2      2      2      2  3.0  1.0  3.0  3.0      2       2   

   Q18 11  Q18 12  Q18 13  Q18 14  Q18 2  Q18 3  Q18 4  Q18 5  Q18 6  Q18 7  \
0       2       2       2       2      2      2      2      2      2      2   
1       2       2       2       2      2      2      2      2      2      2   
2       2       2       2       2      2      2      2      2      2      2   
3       2       2       2       2      2      2      2      2      2      2   
4       2       2       2       2      2      2      2      2      2      2   

   Q18 8  Q18 9  Q18 Disability  Q19 1  Q19 2  Q19 3  Q19 4  Q19 5  Q19 6  \
0      2      2               2      1      2      2      2      2      2   
1      2      2               2      1      2      2      2      2      2   
2      2      2               2      1      2      2      2      2      2   
3      2      2               2      1      2      2      2      2      2   
4      2      2               2      1      2      2      2      2      2   

   Q19 7  Q19 8    Q2  Q2 Cohorts  Q2 Lifecycle  Q20 1  Q20 2  Q20 3  Q20 4  \
0      2      2  35.0         8.0           5.0      2      1      2      2   
1      2      2   3.0         1.0           1.0      2      2      2      2   
2      2      2  36.0         8.0           5.0      2      2      2      2   
3      2      2  32.0         7.0           4.0      2      2      2      2   
4      2      2  32.0         7.0           4.0      2      2      2      2   

   Q20 5  Q20 6  Q20 7  Q20 8  Q20 9  Q21  Q22  Q23  Q24  Q25 1  Q25 2  Q25 3  \
0      2      1      2      2      2  4.0  NaN  2.0  2.0      1      2      2   
1      2      2      2      2      2  NaN  NaN  2.0  NaN      2      2      2   
2      2      2      2      2      2  5.0  NaN  2.0  2.0      1      2      2   
3      2      2      2      2      2  5.0  8.0  1.0  9.0      2      2      2   
4      2      2      2      2      2  3.0  NaN  2.0  1.0      1      2      2   

   Q25 4  Q25 5  Q25 6  Q25 7  Q25 9  Q28     Q29  Q30  Q31  Q32  Q33  Q34 1  \
0      2      2      2      2      2  5.0  3095.0  2.0  4.0  5.0  NaN      2   
1      2      2      2      2      2  NaN     NaN  NaN  NaN 

#For Q9. How long has the person lived at this address?

In [ ]:
person_df.Q9.value_counts()

4.0    617
2.0    426
3.0    278
1.0    198
Name: Q9, dtype: int64

In [ ]:
person_df.Q9.isna().sum() # 26 NAs

26

Following values will be assigned to each duration of stay:

| Length of stay | Scale |
| ----------- | -------- |
| Less than 1 year | 0.5
| 1 to 5 years | 1.5
| 5 to 10 years | 7.5
| 10 years or more | 15
| Nulls | median of the column

In [ ]:
# Generating conditions for each duration of stay and values to be assigned
conditions = [person_df['Q9'].eq(1.0), person_df['Q9'].eq(2.0), person_df['Q9'].eq(3.0), person_df['Q9'].eq(4.0)]
values = [0.5,1.5,7.5,15]

In [ ]:
# Substituting the nan values with median
person_df['Q9'] = np.select(conditions, values, default=person_df['Q9'].median())

Verifying that the values get assigned properly as in the 4th and 5th last records there were nan values. We can see 3 is the median value.

In [ ]:
person_df['Q9'].tail(10) 

1535     0.5
1536     0.5
1537    15.0
1538    15.0
1539    15.0
1540     3.0
1541     3.0
1542     1.5
1543     1.5
1544     1.5
Name: Q9, dtype: float64

#Q10: If less than 5 yrs at this address, where did the person live previously?

In [ ]:
person_df['Q10'].isna().sum()

949

949 missing values. This column will be removed from our analysis.

In [ ]:
person_df = person_df.drop(["Q10"],axis = 1)

#Q15: Does the person spend any time caring for a child / children (under 15 years) without pay?

In [ ]:
person_df['Q15'].isna().sum()  # 68 NAs

68

In [ ]:
person_df['Q15'].head() 

0    1.0
1    5.0
2    1.0
3    5.0
4    1.0
Name: Q15, dtype: float64

In [ ]:
person_df['Q15'].value_counts()

5.0    1071
1.0     262
2.0     104
3.0      26
4.0      14
Name: Q15, dtype: int64

Majority do not spend time caring for children. Mainly people with own children, *262* and grandchildren, *104*. 

In [ ]:
# Need to create separate categories including a new one for nans or missing values. Same as Q3
person_df['Q15'] = pd.to_numeric(person_df['Q15'],errors='ignore').astype(pd.Int64Dtype())
#Creating indicator vars
dummy_vars = pd.get_dummies(person_df["Q15"],prefix="Q15",prefix_sep="_",dummy_na=True)
dummy_vars.head()

Q15_1  Q15_2  Q15_3  Q15_4  Q15_5  Q15_nan
0      1      0      0      0      0        0
1      0      0      0      0      1        0
2      1      0      0      0      0        0
3      0      0      0      0      1        0
4      1      0      0      0      0        0

In [ ]:
person_df = person_df.drop(["Q15"],axis = 1)
person_df = person_df.join(dummy_vars)
person_df.head()

Q1 Selfdescribe Q13 Other Q18 Other Q24 Other               Q26      Q27  \
0             NaN       NaN       NaN       NaN  Aged care worker   Health   
1             NaN       NaN       NaN       NaN               NaN      NaN   
2             NaN       NaN       NaN       NaN       Taxi driver  Tourism   
3             NaN       NaN       NaN       NaN               NaN      NaN   
4             NaN       NaN       NaN       NaN      Truck Driver    Waste   

  Q30 Other Q34 Other Q35 Other Q36 Other  No of person  Person no  Precinct  \
0       NaN       NaN       NaN       NaN             4          1      11.0   
1       NaN       NaN       NaN       NaN             4          2      11.0   
2       NaN       NaN       NaN       NaN             4          3      11.0   
3       NaN       NaN       NaN       NaN             4          4      11.0   
4       NaN       NaN       NaN       NaN             3          1      10.0   

    Q1  Q11  Q12  Q13 1  Q13 10  Q13 11  Q13 12  Q13 2  Q13 3  Q13 4  Q13 5  \
0  2.0  3.0  NaN      2       2       2       2      2      2      2      2   
1  2.0  3.0  NaN      2       2       2       2      2      2      2      2   
2  1.0  3.0  NaN      2       2       2       2      2      2      2      2   
3  1.0  2.0  NaN      2       2       1       2      2      2      2      2   
4  1.0  3.0  NaN      2       2       2       2      2      2      2      2   

   Q13 6  Q13 7  Q13 8  Q13 9  Q14  Q16  Q17  Q18 1  Q18 10  Q18 11  Q18 12  \
0      2      2      2      2  1.0  4.0  3.0      2       2       2       2   
1      2      2      2      2  NaN  6.0  3.0      2       2       2       2   
2      2      2      2      2  3.0  4.0  3.0      2       2       2       2   
3      2      2      2      2  1.0  3.0  3.0      2       2       2       2   
4      2      2      2      2  3.0  3.0  3.0      2       2       2       2   

   Q18 13  Q18 14  Q18 2  Q18 3  Q18 4  Q18 5  Q18 6  Q18 7  Q18 8  Q18 9  \
0       2       2      2      2      2      2      2      2      2      2   
1       2       2      2      2      2      2      2      2      2      2   
2       2       2      2      2      2      2      2      2      2      2   
3       2       2      2      2      2      2      2      2      2      2   
4       2       2      2      2      2      2      2      2      2      2   

   Q18 Disability  Q19 1  Q19 2  Q19 3  Q19 4  Q19 5  Q19 6  Q19 7  Q19 8  \
0               2      1      2      2      2      2      2      2      2   
1               2      1      2      2      2      2      2      2      2   
2               2      1      2      2      2      2      2      2      2   
3               2      1      2      2      2      2      2      2      2   
4               2      1      2      2      2      2      2      2      2   

     Q2  Q2 Cohorts  Q2 Lifecycle  Q20 1  Q20 2  Q20 3  Q20 4  Q20 5  Q20 6  \
0  35.0         8.0           5.0      2      1      2      2      2      1   
1   3.0         1.0           1.0      2      2      2      2      2      2   
2  36.0         8.0           5.0      2      2      2      2      2      2   
3  32.0         7.0           4.0      2      2      2      2      2      2   
4  32.0         7.0           4.0      2      2      2      2      2      2   

   Q20 7  Q20 8  Q20 9  Q21  Q22  Q23  Q24  Q25 1  Q25 2  Q25 3  Q25 4  Q25 5  \
0      2      2      2  4.0  NaN  2.0  2.0      1      2      2      2      2   
1      2      2      2  NaN  NaN  2.0  NaN      2      2      2      2      2   
2      2      2      2  5.0  NaN  2.0  2.0      1      2      2      2      2   
3      2      2      2  5.0  8.0  1.0  9.0      2      2      2      2      2   
4      2      2      2  3.0  NaN  2.0  1.0      1      2      2      2      2   

   Q25 6  Q25 7  Q25 9  Q28     Q29  Q30  Q31  Q32  Q33  Q34 1  Q34 2  Q34 3  \
0      2      2      2  5.0  3095.0  2.0  4.0  5.0  NaN      2      2      2   
1      2      2      2  NaN     NaN  NaN  NaN  5.0  Na

#Q18: Does the person have a permanent or long term disability?

In [ ]:
person_df['Q18 Other'].isna().sum() # 1542 NAs. This column will be dropped.

1542

In [ ]:
person_df.loc[:,'Q18 1':'Q18 Disability'].head()

Q18 1  Q18 10  Q18 11  Q18 12  Q18 13  Q18 14  Q18 2  Q18 3  Q18 4  Q18 5  \
0      2       2       2       2       2       2      2      2      2      2   
1      2       2       2       2       2       2      2      2      2      2   
2      2       2       2       2       2       2      2      2      2      2   
3      2       2       2       2       2       2      2      2      2      2   
4      2       2       2       2       2       2      2      2      2      2   

   Q18 6  Q18 7  Q18 8  Q18 9  Q18 Disability  
0      2      2      2      2               2  
1      2      2      2      2               2  
2      2      2      2      2               2  
3      2      2      2      2               2  
4      2      2      2      2               2

In [ ]:
# Category Vars already exist for Q18, so now simply changing 2s to 0s as the 2s were for un-selected and 1 for a selection
person_df.loc[:,'Q18 1':'Q18 Disability'] = person_df.loc[:,'Q18 1':'Q18 Disability'].applymap(lambda x: 0 if x==2 else x) 

In [ ]:
person_df.loc[:,'Q18 1':'Q18 Disability'].head()

Q18 1  Q18 10  Q18 11  Q18 12  Q18 13  Q18 14  Q18 2  Q18 3  Q18 4  Q18 5  \
0      0       0       0       0       0       0      0      0      0      0   
1      0       0       0       0       0       0      0      0      0      0   
2      0       0       0       0       0       0      0      0      0      0   
3      0       0       0       0       0       0      0      0      0      0   
4      0       0       0       0       0       0      0      0      0      0   

   Q18 6  Q18 7  Q18 8  Q18 9  Q18 Disability  
0      0      0      0      0               0  
1      0      0      0      0               0  
2      0      0      0      0               0  
3      0      0      0      0               0  
4      0      0      0      0               0

In [ ]:
person_df = person_df.drop(["Q18 Other"],axis = 1)

#Q21: If the person is aged 15 years and over, what is the highest qualification the person has attained since leaving school?

In [ ]:
person_df['Q21'].isna().sum() # 296 NAs

296

In [ ]:
person_df['Q21'].value_counts()

1.0    341
5.0    227
4.0    181
2.0    159
3.0    143
6.0    136
9.0     62
Name: Q21, dtype: int64

*   No further education - 341 people
*   Bachelor Degree - 227
*   Diploma - 181
*   Trade certificate - 159
*   Postgraduate - 136


In [ ]:
person_df['Q21'].head() # first 5 records

0    4.0
1    NaN
2    5.0
3    5.0
4    3.0
Name: Q21, dtype: float64

In [ ]:
# Creating dummy vars for catagorical data including creating one for missing values
person_df['Q21'] = pd.to_numeric(person_df['Q21'],errors='ignore').astype(pd.Int64Dtype())
dummy_vars = pd.get_dummies(person_df["Q21"],prefix="Q21",prefix_sep="_",dummy_na=True)

In [ ]:
# Drop Q21 original column
person_df = person_df.drop(["Q21"],axis = 1)
person_df = person_df.join(dummy_vars)
person_df.loc[:,'Q21_1':].head()

Q21_1  Q21_2  Q21_3  Q21_4  Q21_5  Q21_6  Q21_9  Q21_nan
0      0      0      0      1      0      0      0        0
1      0      0      0      0      0      0      0        1
2      0      0      0      0      1      0      0        0
3      0      0      0      0      1      0      0        0
4      0      0      1      0      0      0      0        0

#Q22: If the person attends an educational institution, which type do they attend?

In [ ]:
person_df['Q22'].isna().sum() # 1033 NAs

1033

In [ ]:
person_df['Q22'].value_counts()

9.0     119
5.0      80
8.0      66
2.0      61
10.0     54
4.0      35
1.0      33
6.0      30
7.0      18
3.0      16
Name: Q22, dtype: int64

Categories 1 to 7 (Primary and Seconday) will be combined to one called 'school', 8 to 10 (Tafe, University and other) will be combined to 'uni' and a separate one for missing values, 'nan'.

In [ ]:
person_df['Q22'].head()

0    NaN
1    NaN
2    NaN
3    8.0
4    NaN
Name: Q22, dtype: float64

In [ ]:
# Creating school and uni as categories with values from 1-7 and 8-10 
new_cat = {'school':[1.0,2.0,3.0,4.0,5.0,6.0,7.0], 'uni':[8.0,9.0,10.0]}
mapped = {a:b for b,a in pd.Series(new_cat).explode().iteritems()}

Once categories have been merged we can check that we have school and uni in place of number codes 1 to 7 and 8 to 10.

In [ ]:
person_df['Q22'] = person_df['Q22'].map(mapped).astype('category')
person_df['Q22'].tail()

1540       NaN
1541       NaN
1542       NaN
1543       NaN
1544    school
Name: Q22, dtype: category
Categories (2, object): ['school', 'uni']

In [ ]:
# Creating dummy variables
dummy_vars = pd.get_dummies(person_df["Q22"],prefix="Q22",prefix_sep="_",dummy_na=True)
dummy_vars.head()

Q22_school  Q22_uni  Q22_nan
0           0        0        1
1           0        0        1
2           0        0        1
3           0        1        0
4           0        0        1

In [ ]:
person_df = person_df.drop(['Q22'],axis = 1) # drop Q22 original field

#Q23: Is the person an international student?

In [ ]:
person_df['Q23'].head(20)

0     2.0
1     2.0
2     2.0
3     1.0
4     2.0
5     2.0
6     2.0
7     2.0
8     2.0
9     2.0
10    2.0
11    2.0
12    2.0
13    2.0
14    2.0
15    2.0
16    2.0
17    2.0
18    NaN
19    NaN
Name: Q23, dtype: float64

In [ ]:
person_df['Q23'].isna().sum() # 217 NAs

217

In this field 217 people did not respond and 1318 responded as No. We need to set both to 0s.

In [ ]:
person_df['Q23'].value_counts() # 1318 opted 2

2.0    1318
1.0      10
Name: Q23, dtype: int64

In [ ]:
# Value of 2 will be replaced by 0s, including the missing values or NaNs to be set to 0 as well
person_df.loc[:,'Q23':'Q23'] = person_df.loc[:,'Q23':'Q23'].applymap(lambda x: 0 if x==2 else x).fillna(0)
person_df['Q23'].head(20)

0     0.0
1     0.0
2     0.0
3     1.0
4     0.0
5     0.0
6     0.0
7     0.0
8     0.0
9     0.0
10    0.0
11    0.0
12    0.0
13    0.0
14    0.0
15    0.0
16    0.0
17    0.0
18    0.0
19    0.0
Name: Q23, dtype: float64

#Q33: On average, how long is the person's trips by bicycle (total both ways per trip)?

In [ ]:
person_df['Q33'].head()

0   NaN
1   NaN
2   NaN
3   NaN
4   NaN
Name: Q33, dtype: float64

In [ ]:
person_df['Q33'].isna().sum() # 1215 NAs

1215

In [ ]:
person_df['Q33'].value_counts()

2.0    115
3.0    100
1.0     59
4.0     39
6.0      9
5.0      8
Name: Q33, dtype: int64

Most people use bicycles between 15 to 29 mins (115 people). 100 have responded as between 30 to 59 mins.


In [ ]:
# Keep one column with mid-point values for all 6 options selected
conditions = [person_df['Q33'].eq(1.0), person_df['Q33'].eq(2.0), person_df['Q33'].eq(3.0), person_df['Q33'].eq(4.0), person_df['Q33'].eq(5.0), person_df['Q33'].eq(6.0)]
values = [7.5, 22.5, 45, 74.5, 104.5, 180]
person_df['Q33'] = np.select(conditions, values, default=np.NaN)


Also replace NAs with 0s. The missing values are for people who do not bike.

In [ ]:
person_df['Q33'].fillna(0, inplace=True)

In [ ]:
person_df['Q33'].head(15)

0      0.0
1      0.0
2      0.0
3      0.0
4      0.0
5      0.0
6      0.0
7      0.0
8      0.0
9      0.0
10     0.0
11     0.0
12    45.0
13     0.0
14    45.0
Name: Q33, dtype: float64

#Q37: How safe does the person feel in the public areas of the municipality during the day?

In [ ]:
person_df['Q37'].head()

0    2.0
1    2.0
2    2.0
3    2.0
4    1.0
Name: Q37, dtype: float64

In [ ]:
person_df['Q37'].isna().sum() # 164 NAs

164

In [ ]:
person_df['Q37'].value_counts() 

1.0    640
2.0    527
3.0    139
4.0     48
5.0     27
Name: Q37, dtype: int64

In [ ]:
# Creating dummy vars for catagorical data including creating one for missing values
person_df['Q37'] = pd.to_numeric(person_df['Q37'],errors='ignore').astype(pd.Int64Dtype())
dummy_vars = pd.get_dummies(person_df['Q37'],prefix='Q37',prefix_sep="_",dummy_na=True)

In [ ]:
# Drop Q37 original column
person_df = person_df.drop(['Q37'],axis = 1)
person_df = person_df.join(dummy_vars)
person_df.loc[:,'Q37_1':].head()

Q37_1  Q37_2  Q37_3  Q37_4  Q37_5  Q37_nan
0      0      1      0      0      0        0
1      0      1      0      0      0        0
2      0      1      0      0      0        0
3      0      1      0      0      0        0
4      1      0      0      0      0        0

#Q40: In the last 12 months, has any member of your household accessed or required access to any of the following services?

In [ ]:
household_df.head(2)

Q40 1 Reason Q40 10 Reason Q40 11 Reason Q40 12 Reason Q40 13 Reason  \
0          NaN           NaN           NaN           NaN           NaN   
1          NaN           NaN           NaN           NaN           NaN   

  Q40 14 Reason Q40 15 Reason Q40 16 Reason Q40 17 Reason Q40 18 Name  \
0           NaN           NaN           NaN           NaN         NaN   
1           NaN           NaN           NaN           NaN         NaN   

  Q40 18 Reason Q40 19 Name  Q40 19 Reason Q40 2 Reason Q40 3 Reason  \
0           NaN         NaN            NaN          NaN          NaN   
1           NaN         NaN            NaN          NaN          NaN   

  Q40 4 Reason Q40 5 Reason Q40 6 Reason Q40 7 Reason Q40 8 Reason  \
0          NaN          NaN          NaN          NaN          NaN   
1          NaN          NaN          NaN          NaN          NaN   

  Q40 9 Reason                                         Q41 1  \
0          NaN                   Not good parks for the kids   
1          NaN  Whittlease Township being Metro not regional   

              Q41 2            Q41 3 Q43 Life 1 Rea Q43 Life 2 Rea  \
0  Public transport  Medical centres            NaN            NaN   
1        Road works              NaN            NaN            NaN   

  Q43 Life 3 Rea Q43 Life 4 Rea  \
0            NaN            NaN   
1            NaN            NaN   

                                       Q43 Loc 1 Rea Q43 Loc 2 Rea  \
0                                                NaN           NaN   
1  Lack of public transport and roadworks make it...           NaN   

  Q43 Loc 3 Rea Q43 Loc 4 Rea Q43 Loc 5 Rea Q43 Loc 6 Rea Q43 Loc 7 Rea  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   

                            Q43 Loc 8 Rea Q43 Ser 1 Rea Q43 Ser 2 Rea  \
0  Very long grass near the walking paths           NaN           NaN   
1                                     NaN           NaN           NaN   

  Q43 Ser 3 Rea Q43 Ser 4 Rea Q43 Ser 5 Rea Q43 Ser 6 Rea Q48 Current Other  \
0           NaN           NaN           NaN           NaN               NaN   
1           NaN           NaN           NaN           NaN               NaN   

  Q48 Preferred Other Q49 Other Q53 Other  \
0                 NaN       NaN       NaN   
1                 NaN       NaN       NaN   

                                               Q55 1 Q55 2 Q55 3 Q55 4 Q55 5  \
0  Please keep park area clean and grass free. So...   NaN   NaN   NaN   NaN   
1                                                NaN   NaN   NaN   NaN   NaN   

  Q55 6 Q55 7 Q55 8  Adults 18yrs over  Expanded structure  \
0   NaN   NaN   NaN                3.0                 9.0   
1   NaN   NaN   NaN                2.0                 1.0   

   Household structure  Lote  No of person  Precinct  Q40 1 Access  \
0                    9   1.0             4      11.0             2   
1                    1   2.0             3      10.0             1   

   Q40 1 Difficulty  Q40 1 Notaccess  Q40 10 Access  Q40 10 Difficulty  \
0                 1                2              2                  2   
1                 2                2              2                  2   

   Q40 10 Notaccess  Q40 11 Access  Q40 11 Difficulty  Q40 11 Notaccess  \
0                 2              2                  2                 2   
1                 2              2                  2                 2   

   Q40 12 Access  Q40 12 Diffculty  Q40 12 Notaccess  Q40 13 Access  \
0              2                 2                 2              2   
1              2                 2                 2              1   

   Q40 13 Difficulty  Q40 13 Notaccess  Q40 14 Access  Q40 14 Difficulty  \
0                  2                 2              2                  2   
1                  2                 2              2                  2   

   Q40 14 Notaccess  Q40 15 Access  Q40 15 Difficulty  Q40 

In [ ]:
household_df['Q40 1 Reason'].isna().sum() 

582

In [ ]:
household_df['Q40 1 Reason'].value_counts(dropna = False)

NaN                                                    582
Covid                                                    3
Waiting times                                            2
Distance                                                 2
Waiting time                                             2
Overtimes                                                1
Private hospital                                         1
Used private health insurance                            1
Waiting time are too long                                1
had to go private due to waiting list                    1
Wating time, distance                                    1
Very far to children hospital                            1
Far away from home                                       1
Child hospital - disance                                 1
Wait times                                               1
Closest hospital 30km                                    1
Tests                                                   

There are many NAs in all Q40 reason columns, so they will all be dropped.

In [ ]:
household_df['Q40 2 Reason'].isna().sum() 

585

In [ ]:
household_df['Q40 2 Reason'].value_counts(dropna = False)

NaN                                                    585
Covid                                                    2
Hard to get appointment when really need one.            1
Not in Whittlesea                                        1
Difficult to get an appointment on the day               1
Waiting time are too long                                1
Waiting time                                             1
No appointment availiable                                1
Closed due to Covid19                                    1
Didn't accept new patents                                1
1                                                        1
Transport                                                1
Covid wait time and appointment were not available       1
Covid Booked Out                                         1
Wife was in a nursing home                               1
Opening hours                                            1
Covid restrictions                                      

In [ ]:
cols = ['Q40 1 Reason','Q40 2 Reason','Q40 3 Reason', 'Q40 4 Reason', 'Q40 5 Reason', 'Q40 6 Reason', 'Q40 7 Reason', 'Q40 8 Reason', 'Q40 9 Reason', 'Q40 10 Reason', 'Q40 11 Reason', 'Q40 12 Reason', 'Q40 13 Reason', 'Q40 14 Reason', 'Q40 15 Reason', 'Q40 16 Reason', 'Q40 17 Reason', 'Q40 18 Reason', 'Q40 19 Reason', 'Q40 18 Name', 'Q40 19 Name' ]

In [ ]:
household_df = household_df.drop(cols, axis=1)

In [ ]:
household_df.head(2)

Q41 1             Q41 2  \
0                   Not good parks for the kids  Public transport   
1  Whittlease Township being Metro not regional        Road works   

             Q41 3 Q43 Life 1 Rea Q43 Life 2 Rea Q43 Life 3 Rea  \
0  Medical centres            NaN            NaN            NaN   
1              NaN            NaN            NaN            NaN   

  Q43 Life 4 Rea                                      Q43 Loc 1 Rea  \
0            NaN                                                NaN   
1            NaN  Lack of public transport and roadworks make it...   

  Q43 Loc 2 Rea Q43 Loc 3 Rea Q43 Loc 4 Rea Q43 Loc 5 Rea Q43 Loc 6 Rea  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   

  Q43 Loc 7 Rea                           Q43 Loc 8 Rea Q43 Ser 1 Rea  \
0           NaN  Very long grass near the walking paths           NaN   
1           NaN                                     NaN           NaN   

  Q43 Ser 2 Rea Q43 Ser 3 Rea Q43 Ser 4 Rea Q43 Ser 5 Rea Q43 Ser 6 Rea  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   

  Q48 Current Other Q48 Preferred Other Q49 Other Q53 Other  \
0               NaN                 NaN       NaN       NaN   
1               NaN                 NaN       NaN       NaN   

                                               Q55 1 Q55 2 Q55 3 Q55 4 Q55 5  \
0  Please keep park area clean and grass free. So...   NaN   NaN   NaN   NaN   
1                                                NaN   NaN   NaN   NaN   NaN   

  Q55 6 Q55 7 Q55 8  Adults 18yrs over  Expanded structure  \
0   NaN   NaN   NaN                3.0                 9.0   
1   NaN   NaN   NaN                2.0                 1.0   

   Household structure  Lote  No of person  Precinct  Q40 1 Access  \
0                    9   1.0             4      11.0             2   
1                    1   2.0             3      10.0             1   

   Q40 1 Difficulty  Q40 1 Notaccess  Q40 10 Access  Q40 10 Difficulty  \
0                 1                2              2                  2   
1                 2                2              2                  2   

   Q40 10 Notaccess  Q40 11 Access  Q40 11 Difficulty  Q40 11 Notaccess  \
0                 2              2                  2                 2   
1                 2              2                  2                 2   

   Q40 12 Access  Q40 12 Diffculty  Q40 12 Notaccess  Q40 13 Access  \
0              2                 2                 2              2   
1              2                 2                 2              1   

   Q40 13 Difficulty  Q40 13 Notaccess  Q40 14 Access  Q40 14 Difficulty  \
0                  2                 2              2                  2   
1                  2                 2              2                  2   

   Q40 14 Notaccess  Q40 15 Access  Q40 15 Difficulty  Q40 15 Notaccess  \
0                 2              2                  2                 2   
1                 2              2                  2                 2   

   Q40 16 Access  Q40 16 Difficulty  Q40 16 Notaccess  Q40 17 Access  \
0              2                  2                 1              2   
1              2                  2                 2              2   

   Q40 17 Difficulty  Q40 17 Notaccess  Q40 18 Access  Q40 18 Difficulty  \
0                  2                 2              2                  2   
1                  2                 2              2                  2   

   Q40 18 Notaccess  Q40 19 Access  Q40 19 Difficulty  Q40 19 Notaccess  \
0                 2              2                  2                 2   
1                 2              2                  2                 2   

   Q40 2 Access  Q40 2 Difficulty  Q40 2 Notaccess  Q40 3 Access  \
0             2                 1                2         

Now to handle columns from Q40 1 Access, Q40 1 Difficulty, Q40 1 Notaccess and so on, we need to replace 2s with 0s.



In [ ]:
# Category Vars already exist for Q40, so now simply changing 2s to 0s as the 2s were for un-selected and 1 for a selection
household_df.loc[:,'Q40 1 Access':'Q40 9 Notaccess'] = household_df.loc[:,'Q40 1 Access':'Q40 9 Notaccess'].applymap(lambda x: 0 if x==2 else x) 

In [ ]:
household_df.loc[:,'Q40 1 Access':'Q40 9 Notaccess'].head()

Q40 1 Access  Q40 1 Difficulty  Q40 1 Notaccess  Q40 10 Access  \
0             0                 1                0              0   
1             1                 0                0              0   
2             0                 0                0              0   
3             0                 0                0              0   
4             0                 0                0              0   

   Q40 10 Difficulty  Q40 10 Notaccess  Q40 11 Access  Q40 11 Difficulty  \
0                  0                 0              0                  0   
1                  0                 0              0                  0   
2                  0                 0              0                  0   
3                  0                 0              0                  0   
4                  0                 0              0                  0   

   Q40 11 Notaccess  Q40 12 Access  Q40 12 Diffculty  Q40 12 Notaccess  \
0                 0              0                 0                 0   
1                 0              0                 0                 0   
2                 0              0                 0                 0   
3                 0              0                 0                 0   
4                 0              0                 0                 0   

   Q40 13 Access  Q40 13 Difficulty  Q40 13 Notaccess  Q40 14 Access  \
0              0                  0                 0              0   
1              1                  0                 0              0   
2              0                  0                 0              0   
3              0                  0                 0              0   
4              0                  0                 0              0   

   Q40 14 Difficulty  Q40 14 Notaccess  Q40 15 Access  Q40 15 Difficulty  \
0                  0                 0              0                  0   
1                  0                 0              0                  0   
2                  0                 0              0                  0   
3                  0                 0              0                  0   
4                  0                 0              0                  0   

   Q40 15 Notaccess  Q40 16 Access  Q40 16 Difficulty  Q40 16 Notaccess  \
0                 0              0                  0                 1   
1                 0              0                  0                 0   
2                 0              0                  0                 0   
3                 0              0                  0                 0   
4                 0              0                  0                 0   

   Q40 17 Access  Q40 17 Difficulty  Q40 17 Notaccess  Q40 18 Access  \
0              0                  0                 0              0   
1              0                  0                 0              0   
2              0                  0                 0              0   
3              0                  0                 0              0   
4              0                  0                 0              0   

   Q40 18 Difficulty  Q40 18 Notaccess  Q40 19 Access  Q40 19 Difficulty  \
0                  0                 0              0                  0   
1                  0                 0              0                  0   
2                  0                 0              0                  0   
3                  0                 0              0                  0   
4                  0                 0              0                  0   

   Q40 19 Notaccess  Q40 2 Access  Q40 2 Difficulty  Q40 2 Notaccess  \
0                 0             0                 1                0   
1                 0             0                 1                0   
2                 0             1                 0                0   
3                 0             1                 0                0   
4                 0             1                 0                0   

   Q40 3 Access  Q40 3

#Q42: On a scale of 0 (strongly disagree) to 10 (strongly agree), to what extent do you agree or disagree that there are suitable local education options available to meet your household's needs?

In [ ]:
household_df.loc[:,'Q42 1':'Q42 6'].head(6)

Q42 1  Q42 2  Q42 3  Q42 4  Q42 5  Q42 6
0    5.0    0.0    5.0    0.0    0.0    0.0
1    3.0    4.0    4.0    NaN    NaN    NaN
2    5.0    4.0    6.0    NaN    NaN    NaN
3    NaN    NaN    NaN    NaN    NaN    NaN
4    NaN    NaN    NaN    NaN    NaN    NaN
5    NaN    NaN    NaN    NaN    NaN    NaN

In [ ]:
household_df.loc[:,'Q42 1':'Q42 6'].isna().sum() # Over 300 NAs in each category

Q42 1    315
Q42 2    306
Q42 3    312
Q42 4    346
Q42 5    324
Q42 6    360
dtype: int64

In [ ]:
household_df.loc[:,'Q42 1':'Q42 6'].apply(pd.Series.value_counts)

Q42 1  Q42 2  Q42 3  Q42 4  Q42 5  Q42 6
0.0       8      6      7     18     29     22
1.0       3      6     12     13     13     15
2.0      11      6     17     22     23     21
3.0       8     11     20     14     17     22
4.0       6      9     10     12     11     10
5.0      69     65     61     71     66     71
6.0      12     14     17     12     17     14
7.0      17     24     19     21     16     13
8.0      51     48     45     29     28     15
9.0      48     48     33     20     23     13
10.0     60     65     55     30     41     32

0, 1, 2, 3 - Strongly disagree
4, 5, 6  - Neutral
7, 8, 9, 10 - Strongly agree
1. Q42 1 - Early Years 
2. Q42 2 - Primary
3. Q42 3 - Secondary 
4. Q42 4 - TAFE
5. Q42 5 - University/Similar
6. Q42 6 - Adult Education
*Majority* people (69) are 'neutral' on Early years education but good numer (66) strongly agree on Primary education availability. All NAs in the Q42 6 categories will be replaced by Neutral i.e. level 5.

In [ ]:
# Filling all NAs with 5 (which is neutral)
household_df.loc[:,'Q42 1':'Q42 6'] = household_df.loc[:,'Q42 1':'Q42 6'].fillna(5)

In [ ]:
# Converting it to numeric from floats 
household_df.loc[:,'Q42 1':'Q42 6'] = household_df.loc[:,'Q42 1':'Q42 6'].astype(int)
household_df.loc[:,'Q42 1':'Q42 6'].head()

Q42 1  Q42 2  Q42 3  Q42 4  Q42 5  Q42 6
0      5      0      5      0      0      0
1      3      4      4      5      5      5
2      5      4      6      5      5      5
3      5      5      5      5      5      5
4      5      5      5      5      5      5

#Q45: What are all the reasons members of your household visit local open spaces?

In [ ]:
household_df.loc[:,'Q45 1':'Q45 9'].isna().sum() # No NAs

Q45 1     0
Q45 10    0
Q45 11    0
Q45 12    0
Q45 13    0
Q45 2     0
Q45 3     0
Q45 4     0
Q45 5     0
Q45 6     0
Q45 7     0
Q45 8     0
Q45 9     0
dtype: int64

In [ ]:
household_df.loc[:,'Q45 1':'Q45 9'].head()

Q45 1  Q45 10  Q45 11  Q45 12  Q45 13  Q45 2  Q45 3  Q45 4  Q45 5  Q45 6  \
0      1       2       2       2       2      1      2      2      1      2   
1      1       2       2       2       2      2      1      2      1      1   
2      1       2       1       1       2      1      1      1      1      1   
3      2       2       2       2       2      2      2      2      2      2   
4      2       2       2       2       2      2      2      2      2      2   

   Q45 7  Q45 8  Q45 9  
0      2      1      1  
1      2      1      1  
2      2      1      2  
3      2      2      2  
4      2      2      2

Replacing 2s with 0s.

In [ ]:
# Category Vars already exist for Q45, so now simply changing 2s to 0s as the 2s were for un-selected and 1 for a selection
household_df.loc[:,'Q45 1':'Q45 9'] = household_df.loc[:,'Q45 1':'Q45 9'].applymap(lambda x: 0 if x==2 else x) 

In [ ]:
household_df.loc[:,'Q45 1':'Q45 9'].head()

Q45 1  Q45 10  Q45 11  Q45 12  Q45 13  Q45 2  Q45 3  Q45 4  Q45 5  Q45 6  \
0      1       0       0       0       0      1      0      0      1      0   
1      1       0       0       0       0      0      1      0      1      1   
2      1       0       1       1       0      1      1      1      1      1   
3      0       0       0       0       0      0      0      0      0      0   
4      0       0       0       0       0      0      0      0      0      0   

   Q45 7  Q45 8  Q45 9  
0      0      1      1  
1      0      1      1  
2      0      1      0  
3      0      0      0  
4      0      0      0

#Q50: What is the home loan repayment or rent payment on this dwelling?

In [ ]:
household_df['Q50'].head()

0    6.0
1    4.0
2    4.0
3    NaN
4    NaN
Name: Q50, dtype: float64

In [ ]:
household_df['Q50'].isna().sum() # 338 NAs

337

In [ ]:
household_df['Q50'].value_counts(dropna=False)

NaN    337
5.0     76
6.0     72
4.0     71
3.0     28
1.0     15
2.0      9
Name: Q50, dtype: int64

Setting mid-point values for loan or rent repayment on the dwelling, 6 will become 750, so 5 will become 450, 4 will be 350, 3 will become 250, 2 will be 150, 1 will be 50 and all NAs will be set to 0. 

In [ ]:
conditions = [household_df['Q50'].eq(1.0), household_df['Q50'].eq(2.0), household_df['Q50'].eq(3.0), household_df['Q50'].eq(4.0), household_df['Q50'].eq(5.0), household_df['Q50'].eq(6.0)]
values = [50, 150, 250, 350, 450, 750]
household_df['Q50'] = np.select(conditions, values, default=0)

Now 338 loan repayments are 0. 

In [ ]:
household_df['Q50'].value_counts(dropna=False)

0      337
450     76
750     72
350     71
250     28
50      15
150      9
Name: Q50, dtype: int64

#Q54: Where does your household shop most often for your shopping / dining needs?

In [ ]:
household_df.head(2)

Q41 1             Q41 2  \
0                   Not good parks for the kids  Public transport   
1  Whittlease Township being Metro not regional        Road works   

             Q41 3 Q43 Life 1 Rea Q43 Life 2 Rea Q43 Life 3 Rea  \
0  Medical centres            NaN            NaN            NaN   
1              NaN            NaN            NaN            NaN   

  Q43 Life 4 Rea                                      Q43 Loc 1 Rea  \
0            NaN                                                NaN   
1            NaN  Lack of public transport and roadworks make it...   

  Q43 Loc 2 Rea Q43 Loc 3 Rea Q43 Loc 4 Rea Q43 Loc 5 Rea Q43 Loc 6 Rea  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   

  Q43 Loc 7 Rea                           Q43 Loc 8 Rea Q43 Ser 1 Rea  \
0           NaN  Very long grass near the walking paths           NaN   
1           NaN                                     NaN           NaN   

  Q43 Ser 2 Rea Q43 Ser 3 Rea Q43 Ser 4 Rea Q43 Ser 5 Rea Q43 Ser 6 Rea  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   

  Q48 Current Other Q48 Preferred Other Q49 Other Q53 Other  \
0               NaN                 NaN       NaN       NaN   
1               NaN                 NaN       NaN       NaN   

                                               Q55 1 Q55 2 Q55 3 Q55 4 Q55 5  \
0  Please keep park area clean and grass free. So...   NaN   NaN   NaN   NaN   
1                                                NaN   NaN   NaN   NaN   NaN   

  Q55 6 Q55 7 Q55 8  Adults 18yrs over  Expanded structure  \
0   NaN   NaN   NaN                3.0                 9.0   
1   NaN   NaN   NaN                2.0                 1.0   

   Household structure  Lote  No of person  Precinct  Q40 1 Access  \
0                    9   1.0             4      11.0             0   
1                    1   2.0             3      10.0             1   

   Q40 1 Difficulty  Q40 1 Notaccess  Q40 10 Access  Q40 10 Difficulty  \
0                 1                0              0                  0   
1                 0                0              0                  0   

   Q40 10 Notaccess  Q40 11 Access  Q40 11 Difficulty  Q40 11 Notaccess  \
0                 0              0                  0                 0   
1                 0              0                  0                 0   

   Q40 12 Access  Q40 12 Diffculty  Q40 12 Notaccess  Q40 13 Access  \
0              0                 0                 0              0   
1              0                 0                 0              1   

   Q40 13 Difficulty  Q40 13 Notaccess  Q40 14 Access  Q40 14 Difficulty  \
0                  0                 0              0                  0   
1                  0                 0              0                  0   

   Q40 14 Notaccess  Q40 15 Access  Q40 15 Difficulty  Q40 15 Notaccess  \
0                 0              0                  0                 0   
1                 0              0                  0                 0   

   Q40 16 Access  Q40 16 Difficulty  Q40 16 Notaccess  Q40 17 Access  \
0              0                  0                 1              0   
1              0                  0                 0              0   

   Q40 17 Difficulty  Q40 17 Notaccess  Q40 18 Access  Q40 18 Difficulty  \
0                  0                 0              0                  0   
1                  0                 0              0                  0   

   Q40 18 Notaccess  Q40 19 Access  Q40 19 Difficulty  Q40 19 Notaccess  \
0                 0              0                  0                 0   
1                 0              0                  0                 0   

   Q40 2 Access  Q40 2 Difficulty  Q40 2 Notaccess  Q40 3 Access  \
0             0                 1                0         

In [ ]:
df = household_df.loc[:,'q54 1 CCG':'q54 9 LHG']

In [ ]:
df.isna().sum() # No NAs

q54 1 CCG     0
q54 1 DN      0
q54 1 DOE     0
q54 1 GS      0
q54 1 LHG     0
q54 10 CCG    0
q54 10 DN     0
q54 10 DOE    0
q54 10 GS     0
q54 10 LHG    0
q54 11 CCG    0
q54 11 DN     0
q54 11 DOE    0
q54 11 GS     0
q54 11 LHG    0
q54 12 CCG    0
q54 12 DN     0
q54 12 DOE    0
q54 12 GS     0
q54 12 LHG    0
q54 13 CCG    0
q54 13 DN     0
q54 13 DOE    0
q54 13 GS     0
q54 13 LHG    0
q54 14 CCG    0
q54 14 DN     0
q54 14 DOE    0
q54 14 GS     0
q54 14 LHG    0
q54 15 CCG    0
q54 15 DN     0
q54 15 DOE    0
q54 15 GS     0
q54 15 LHG    0
q54 16 CCG    0
q54 16 DN     0
q54 16 DOE    0
q54 16 GS     0
q54 16 LHG    0
q54 17 CCG    0
q54 17 DN     0
q54 17 DOE    0
q54 17 GS     0
q54 17 LHG    0
q54 18 CCG    0
q54 18 DN     0
q54 18 DOE    0
q54 18 GS     0
q54 18 LHG    0
q54 19 CCG    0
q54 19 DN     0
q54 19 DOE    0
q54 19 GS     0
q54 19 LHG    0
q54 2 CCG     0
q54 2 DN      0
q54 2 DOE     0
q54 2 GS      0
q54 2 LHG     0
q54 20 CCG    0
q54 20 DN     0
q54 20 D

In [ ]:
household_df.loc[:,'q54 1 CCG':'q54 9 LHG'].apply(pd.Series.value_counts)

q54 1 CCG  q54 1 DN  q54 1 DOE  q54 1 GS  q54 1 LHG  q54 10 CCG  q54 10 DN  \
1          4        44          9        45        NaN           4         45   
2        604       564        599       563      608.0         604        563   

   q54 10 DOE  q54 10 GS  q54 10 LHG  q54 11 CCG  q54 11 DN  q54 11 DOE  \
1          28         73         NaN         NaN         26           1   
2         580        535       608.0       608.0        582         607   

   q54 11 GS  q54 11 LHG  q54 12 CCG  q54 12 DN  q54 12 DOE  q54 12 GS  \
1         21         NaN           2         53           3         60   
2        587       608.0         606        555         605        548   

   q54 12 LHG  q54 13 CCG  q54 13 DN  q54 13 DOE  q54 13 GS  q54 13 LHG  \
1           1           1         60           9         78         NaN   
2         607         607        548         599        530       608.0   

   q54 14 CCG  q54 14 DN  q54 14 DOE  q54 14 GS  q54 14 LHG  q54 15 CCG  \
1           9         51           7         51           1         151   
2         599        557         601        557         607         457   

   q54 15 DN  q54 15 DOE  q54 15 GS  q54 15 LHG  q54 16 CCG  q54 16 DN  \
1          9          63         15          85           2         24   
2        599         545        593         523         606        584   

   q54 16 DOE  q54 16 GS  q54 16 LHG  q54 17 CCG  q54 17 DN  q54 17 DOE  \
1           2         30           3         271         84         230   
2         606        578         605         337        524         378   

   q54 17 GS  q54 17 LHG  q54 18 CCG  q54 18 DN  q54 18 DOE  q54 18 GS  \
1        151          71          17         32          28         41   
2        457         537         591        576         580        567   

   q54 18 LHG  q54 19 CCG  q54 19 DN  q54 19 DOE  q54 19 GS  q54 19 LHG  \
1          10           1          9           7         18          14   
2         598         607        599         601        590         594   

   q54 2 CCG  q54 2 DN  q54 2 DOE  q54 2 GS  q54 2 LHG  q54 20 CCG  q54 20 DN  \
1          5        25         12        40          2           4          6   
2        603       583        596       568        606         604        602   

   q54 20 DOE  q54 20 GS  q54 20 LHG  q54 21 CCG  q54 21 DN  q54 21 DOE  \
1           5         10          91         188         33          41   
2         603        598         517         420        575         567   

   q54 21 GS  q54 21 LHG  q54 22 CCG  q54 22 DN  q54 22 DOE  q54 22 GS  \
1         44          16          14         63          36         51   
2        564         592         594        545         572        557   

   q54 22 LHG  q54 23 CCG  q54 23 DN  q54 23 DOE  q54 23 GS  q54 23 LHG  \
1           1          50          3          90          3          13   
2         607         558        605         518        605         595   

   q54 24 CCG  q54 24 DN  q54 24 DOE  q54 24 GS  q54 24 LHG  q54 25 CCG  \
1           8          9           6         30         NaN         NaN   
2         600        599         602        578       608.0       608.0   

   q54 25 DN  q54 25 DOE  q54 25 GS  q54 25 LHG  q54 26 CCG  q54 26 DN  \
1          1         NaN          8         NaN         122          8   
2        607       608.0        600       608.0         486        600   

   q54 26 DOE  q54 26 GS  q54 26 LHG  q54 3 CCG  q54 3 DN  q54 3 DOE  \
1          13         28          90         36        15         20   
2         595        580         518        572       593        588   

   q54 3 GS  q54 3 LHG  q54 4 CCG  q54 4 DN  q54 4 DOE  q54 4 GS  q54 4 LHG  \
1        19         18        213        71        140       145         62   
2       589        590        395       537        468       463        546   

   q54 5 CCG  q54 5 DN  q54 5 DOE  q54 5 GS  q54 5 LHG  q54 6 CCG  q54 6 DN  \
1          3        12          9         6          2    

In [ ]:
# Category Vars already exist for Q54, so now simply changing 2s to 0s as the 2s were for un-selected and 1 for a selection
household_df.loc[:,'q54 1 CCG':'q54 9 LHG'] = household_df.loc[:,'q54 1 CCG':'q54 9 LHG'].applymap(lambda x: 0 if x==2 else x) 

After changing 2s to 0s. All 2s mean no selection and 1 is a selection. We can see now we have 1s and 0s.

In [ ]:
household_df.loc[:,'q54 1 CCG':'q54 9 LHG'].head(2)

q54 1 CCG  q54 1 DN  q54 1 DOE  q54 1 GS  q54 1 LHG  q54 10 CCG  q54 10 DN  \
0          0         1          0         0          0           0          0   
1          0         0          0         0          0           0          0   

   q54 10 DOE  q54 10 GS  q54 10 LHG  q54 11 CCG  q54 11 DN  q54 11 DOE  \
0           0          0           0           0          0           0   
1           0          1           0           0          0           0   

   q54 11 GS  q54 11 LHG  q54 12 CCG  q54 12 DN  q54 12 DOE  q54 12 GS  \
0          0           0           0          0           0          0   
1          0           0           0          0           0          0   

   q54 12 LHG  q54 13 CCG  q54 13 DN  q54 13 DOE  q54 13 GS  q54 13 LHG  \
0           0           0          0           0          0           0   
1           0           0          0           0          0           0   

   q54 14 CCG  q54 14 DN  q54 14 DOE  q54 14 GS  q54 14 LHG  q54 15 CCG  \
0           0          0           0          0           0           1   
1           0          0           0          0           0           0   

   q54 15 DN  q54 15 DOE  q54 15 GS  q54 15 LHG  q54 16 CCG  q54 16 DN  \
0          0           0          0           0           0          0   
1          0           0          0           0           0          0   

   q54 16 DOE  q54 16 GS  q54 16 LHG  q54 17 CCG  q54 17 DN  q54 17 DOE  \
0           0          0           0           0          0           0   
1           0          0           0           1          0           0   

   q54 17 GS  q54 17 LHG  q54 18 CCG  q54 18 DN  q54 18 DOE  q54 18 GS  \
0          0           0           0          0           0          0   
1          0           0           1          0           1          0   

   q54 18 LHG  q54 19 CCG  q54 19 DN  q54 19 DOE  q54 19 GS  q54 19 LHG  \
0           0           0          0           0          0           0   
1           0           0          0           0          0           0   

   q54 2 CCG  q54 2 DN  q54 2 DOE  q54 2 GS  q54 2 LHG  q54 20 CCG  q54 20 DN  \
0          0         0          0         0          0           0          0   
1          0         0          0         0          0           0          0   

   q54 20 DOE  q54 20 GS  q54 20 LHG  q54 21 CCG  q54 21 DN  q54 21 DOE  \
0           1          0           0           0          0           0   
1           0          0           1           0          0           0   

   q54 21 GS  q54 21 LHG  q54 22 CCG  q54 22 DN  q54 22 DOE  q54 22 GS  \
0          0           0           0          0           0          0   
1          0           0           0          1           0          0   

   q54 22 LHG  q54 23 CCG  q54 23 DN  q54 23 DOE  q54 23 GS  q54 23 LHG  \
0           0           0          0           0          0           0   
1           0           0          0           0          0           0   

   q54 24 CCG  q54 24 DN  q54 24 DOE  q54 24 GS  q54 24 LHG  q54 25 CCG  \
0           0          0           0          0           0           0   
1           0          0           0          0           0           0   

   q54 25 DN  q54 25 DOE  q54 25 GS  q54 25 LHG  q54 26 CCG  q54 26 DN  \
0          0           0          0           0           0          0   
1          0           0          0           0           1          0   

   q54 26 DOE  q54 26 GS  q54 26 LHG  q54 3 CCG  q54 3 DN  q54 3 DOE  \
0           0          0           0          0         0          0   
1           0          0           0          0         0          0   

   q54 3 GS  q54 3 LHG  q54 4 CCG  q54 4 DN  q54 4 DOE  q54 4 GS  q54 4 LHG  \
0         0          0          0         0          0         1          0   
1         0          0          1         0          1         0          0   

   q54 5 CCG  q54 5 DN  q54 5 DOE  q54 5 GS  q54 5 LHG  q54 6 CCG  q54 6 DN  \
0          0         0          0         0          0    

#Q27: In what industry does the person usually work?

Industries:

(i) Agriculture

(ii) Manufacturing

(iii) Utilities

(iv) Construction

(v) Wholesale Trade

(vi) Retail Trade

(vii) Hospitality (Accommodation and Food Services)

(viii) Transportation & Logistics

(ix) Professional and Other Services (Financial, Insurance, IT, Media, Real Estate, Scientific, Technical)

(x) Administrative and Support Services

(xi) Public Administration/Government

(xii) Education, Training and Arts

(xiii) Healthcare and Social Assistance

Data in Q27 can belong to any of the above categories.

In [ ]:
# List of Industries 

industry_list = ['Agriculture', 'Manufacturing', 'Utilities', 'Construction', 'Wholesale Trade', 'Retail Trade', 'Transportation & Logistics', 
                 'Hospitality (Accommodation and Food Services)', 'Healthcare and Social Assistance', 'Education, Training and Arts',
                 'Professional and Other Services (Financial, Insurance, IT, Media, Real Estate, Scientific, Technical)',
                 'Administrative and Support Services', 'Public Administration/Government']

Downloading the category value counts in Q27 to spreadsheet for easier analysis of the data. Sorting the categories in alphabetical order for better viewing.

In [ ]:
df = person_df['Q27'].value_counts().sort_index(ascending=True)
filepath = "/content/Q27_categories.csv"
df.to_csv(filepath)

In [ ]:
person_df['Q27_Original'] = person_df['Q27']

In [ ]:
person_df['Q27_Original'].head()

0     Health
1        NaN
2    Tourism
3        NaN
4      Waste
Name: Q27_Original, dtype: object

In [ ]:
!pip install fuzzywuzzy

In [ ]:
import fuzzywuzzy
from fuzzywuzzy import process

In [ ]:
# Creating a list matches that contains match of each Q27 entry against the industry list created above.
matches=[]

for x in person_df.Q27:
    try:
        matches.append([process.extract(x, industry_list, limit=1)][0][0][0])
    except:
        matches.append(np.nan)

If we just use Fuzzy matching with current strings inside the 'Q27' column and the list of industries, we get reasonable match only for strings that have a match or some pattern.

In [ ]:
person_df['Q27_target'] = matches                                 # created a Q27_target column in person_df
cols = [ 'Q27_Original', 'Q27_target', 'Survey no', 'Person no']  # For easier viewing keeping relevant columns together
person_df[cols].sort_values(by=['Survey no', 'Person no'], ascending=True).head(20) # sort by suvery no for viweing

Q27_Original                                     Q27_target  \
0               Health               Healthcare and Social Assistance   
1                  NaN                                            NaN   
2              Tourism                                  Manufacturing   
3                  NaN                                            NaN   
4                Waste            Administrative and Support Services   
5          Hospitality  Hospitality (Accommodation and Food Services)   
6                  NaN                                            NaN   
7            Insurance               Healthcare and Social Assistance   
8              Teacher                                Wholesale Trade   
9   Hospitality/ Sales  Hospitality (Accommodation and Food Services)   
10                 NaN                                            NaN   
11                 NaN                                            NaN   
12           Aged Care                                    Agriculture   
13                 NaN                                            NaN   
14           Wholesale                                Wholesale Trade   
15                 NaN                                            NaN   
16                 NaN                                            NaN   
17                  IT                                      Utilities   
18           Education                   Education, Training and Arts   
19           Education                   Education, Training and Arts   

    Survey no  Person no  
0           1          1  
1           1          2  
2           1          3  
3           1          4  
4           2          1  
5           2          2  
6           2          3  
7           3          1  
8           3          2  
9           4          1  
10          5          1  
11          5          2  
12          6          1  
13          6          2  
14          7          1  
15          8          1  
16          8          2  
17          8          3  
18          9          1  
19          9          2

As we can see there are some unusual matches Kindergarten has matched to Public Administration/Government, None has become Transportation & Logistics, Hospital has become Hospitality, Aged care matched to Agriculture and so on. To improve this matching, some data massaging is necessary to improve category matching, i.e. we need to pre-process the data so String/Pattern Matching does not work and we may need to hard code the values in a list and use Python's standard pattern matching.

In [ ]:
person_df = person_df.drop(['Q27_target', 'Q27_Original'],axis = 1) # Now we don't need these columns.

Before any pre-processing we need to address NAs and None values in Q27 column. For now we will place them as a placeholder under category 'Missing' 

In [ ]:
#person_df['Q27'].mask(person_df['Q27'] == 'None', 'Missing', inplace=True)
#person_df['Q27'] = person_df['Q27'].fillna('Missing')


As we can see there are some 344 categories in Q27 column. We will now provide data to place in a category (industry). For instance, 
* 'IT', 'Banking' and so on belong to 'Professional and Other services'
* 'Teacher' to 'Education, Training and Arts'
* 'Dental' as 'Healthcare' and so on.

So all data is hard coded in separate occupation lists and Q27 is updated with corresponding category for each input list.



Industry category 1:

Professional and Other Services (Financial, Insurance, IT, Media, Real Estate, Scientific, Technical)

In [ ]:
professions = ['Accounting', 'Accounting Firm','Advertising','All','Animal','Architect','Audio Visual', 'Banking', 'Beauty','Beauty and Hair',
               'Business', 'CBA','CMT', 'CPVV','Cabinet Maker', 'Car sales person','Clerical', 'Cleaning', 'Contract Research', 'Cust Service',
               'Customer Service', 'Customer Service Officer', 'Customer Service Rep', 'Cuvality Control', 'Cyber Security', 'Data Entry',
               'Designer', 'Door Shop', 'Draftsman', 'Driving school', 'Elecrician', 'Electrician', 'Electronic Eng', 'Environmental Consultancy',
               'Engineering', 'Event Rigging and Lighting', 'Events', 'Factory', 'Factory Worker', 'Fashion', 'Fatorie outlet', 'Finace', 
               'Finance', 'Financial', 'Financial planning', 'Fitness', 'Fitness, Entertainment', 'Floorcoverings', 'Garden', 'Gardening',
               'Hairdresser', 'Hairdressing', 'Home', 'House Maker', 'Housing', 'IT', 'IT & Marketing', 'Information Technology',
               'Information technology', 'Insurance', 'Insurance and Finance', 'Labour', 'Laundry hand', 'Legal', 'Library Technician', 'Many',
               'Mechanical', 'Mechanics', 'Media', 'Milk bar', 'Outdoor', 'Outside','Packaging','Payroll','Phone','Photographer','Plumbing', 
               'Printing', 'Private', 'Private Mechanic Company', 'Property', 'Real Estate', 'Religious organisation', 'Residential Homes',
               'Sales', 'Security', 'Security officer', 'Self employed', 'Space sauers', 'Sport', 'Sport support', 'Swan', 'Sysem integration and consulting',
               'Tailor', 'Tattoo', 'Taxi', 'Tech', 'Telecommunication', 'Telecommunications', 'Test assembly operator', 'Trade', 'Trades and service',
               'Tradesman','Varied', 'Veterinary', 'Welder', 'Windows', 'Caterpiller', 'None', 'ST']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Professional and Other Services'

Industry Category 2: Education, Training and Arts

In [ ]:
professions =['Aboriginal Early Childhood','Academic','Art', 'Educatiom', 'Education', 'Entertainment','International Entertainment','Kindergarden',
              'Kindergarden Educator', 'Kingergarden', 'Migher ED', 'Research lab', 'School', 'Schools', 'Sec Education', 'Still At School, Retail (Casual)',
              'Still School', 'Student', 'Teacher', 'University', 'Primary']  
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Education, Training and Arts'

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:5: UserWarning: This pattern has match groups. To actually get the groups, use str.extract.
  """


Industry Category 3: Healthcare and Social Assistance

In [ ]:
professions =['Aboriginal mental Health', 'Age Care Facility', 'Age Carer', 'Agecare', 'Aged Care', 'Aged care', 'Allied Health', 
              'Austin Health mental health service', 'Chemist', 'Child Care', 'Childcare', 'Community health', 'Dental', 'Disability',
              'Disability Employment', 'Disability care', 'Doctors', 'Drug testing', 'Full time carer', 'Healtcare', 'Health', 'Health Care',
              'Health Industry', 'Health care', 'Health/ Aged Care', 'Healthcare', 'In home aged care', 'LGA', 'Medical', 'Medical Admin',
              'Medical Research', 'Medical Science', 'Medical services', 'Mental Health', 'Nursing Home', 'Nursing/ Retired', 'Pharmaceuticals',
              'Public Health', 'Public Hospital', 'Rathology', 'Vet Care']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Healthcare and Social Assistance'

Industry category 4: Transportation & Logistics

In [ ]:
professions =['Airfreight', 'Airline', 'Aviation', 'Cumeal Trial Logistics', 'Freight', 'Freight Forwarding', 'Frieght', 'JB were', 'Logistic & WH', 
              'Logistics', 'Transport', 'Transport & Logistics','Transport and Packing Industry', 'Transport/Logistics', 'Transportation', 
              'Warehosuing Distribution', 'Warehouse','Warehouse/ Dist', 'Warehousing']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Transportation & Logistics'

Industry Category 5: Utilities

In [ ]:
professions =['Energy', 'Minning', 'Oil and Gas', 'Power and Gas', 'Water/Gas', 'Utilities']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Utilities'

Industry Category 6 : Construction

In [ ]:
professions =['Asphalting Supervisor','Bowling construction', 'Buiding Industry', 'Building', 'Building & Construction', 'Building Industry',
              'Building Supplies', 'Building/ Renovations', 'Buildng', 'Civil Construction', 'Civil construction', 'Construction', 
              'Construction Hire', 'Construction Supervisor', 'Constructions', 'Contruction', 'Demolition', 'Home Construction', 'Landscape Design']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Construction'

Industry Category 7 : Hospitality (Accommodation and Food Services)

In [ ]:
professions =['Bakery', 'Catering', 'Factory Food', 'Fast Food', 'Fast Food Indusry', 'Food', 'Food Distribution', 'Food Industry', 'Food and Beverage',
              'Food/Retail', 'Fresh Foods', 'Fruit and Veges', 'Hospatality', 'Hospital', 'Hospitality', 'Hospitality/ Sales', 'Hosptality', 
              'Kitchen', 'McDonalds']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Hospitality (Accommodation and Food Services)'

Industry Category 8 : Manufacturing

In [ ]:
professions =['Automotive', 'Car Industry', 'Food Manufacturing', 'Foundry', 'Industrial', 'Manufactory', 'Manufacturing', 'Manufacturing & Signage Company',
              'Plastics', 'Production', 'Shoes']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Manufacturing'

Industry Category 9 : Agriculture

In [ ]:
professions =['Electrical/ Farming', 'Farming', 'Farmer']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Agriculture'

Industry Category 10 : Public Administration/Government

In [ ]:
professions =['ADF veteran', 'Admin', 'Council', 'Defence', 'Defence Force', 'Defence/ Retired', 'Emergency Management', 'Emergency Responder',
              'Emergency Services', 'Federal Govt', 'Federal Govt Welfare', 'Government', 'Government Department', 'Government Housing',
              'Human Services', 'Justice Dept', 'Law', 'Law Enforcement', 'Local Government', 'Local Govnerment', 'Local laws', 'NDIS', 
              'Police', 'Police Officer', 'Public', 'Public Servant', 'Public Service', 'Public Transport', 'Public servant', 'Public servant - case management',
              'Public services', 'Publiv Victoria Service', 'RTO', 'Signage', 'Signage manufact', 'Social Security', 'State Govt', 'State Research and Development',
              'TPI Pensioner', 'Vic Gov Transport', 'Waste', 'Waste management', 'Whittlesea Council', 'emergency services', 'Qsuper']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Public Administration/Government'

Industry Category 11 : Wholesale Trade

In [ ]:
professions =['Bulk Handling', 'Hardware', 'Securing Wholesale', 'Storage and Wholesale', 'Wholesale']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Wholesale Trade'

Industry Category 12 : Retail Trade

In [ ]:
professions =['Butcher', 'Car retail', 'Car sales', 'Caravan', 'Caravan Industry', 'Cars', 'Casino dealer', 'Clothing', 'Coles', 
              'Commercial Cleaning', 'Commerical Building', 'Department Store', 'Dog Food', 'Electrical', 'Electrical Goods Store',
              'Ford', 'Ford motor Pty Ltd', 'LV Motor Trade', 'Restaurant', 'Resturant', 'Retail', 'Retail Shop', 'Supermaket', 
              'Supermarket', 'Wool']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Retail Trade'

Industry Category 13 : Administrative and Support Services

In [ ]:
professions =['Recognition + Reward','Australia Post', 'Depot', 'Essential', 'Fuel Station', 'Fundraising', 'Kids Playground', 'Maintenances', 'Maritime', 'Non-profit', 
              'Not for Profit','Not for profit', 'Post', 'Postal','Service Industry', 'Taxation', 'Youth', 'Tourism', 'Travel and Tourism']
pattern = '|'.join(professions)
index = person_df[person_df['Q27'].str.contains(pattern, case=True, na=False)].index.values
person_df.loc[index,'Q27'] = 'Administrative and Support Services'
index = person_df[person_df['Q27'].str.contains('Recognition + Reward', case=True, na=False, regex=False)].index.values
person_df.loc[index,'Q27'] = 'Administrative and Support Services'

Let's check the Q27 column again, displaying it's first 10 values. And run value counts again, to ensure all data only belongs to above mentioned categories.

In [ ]:
person_df['Q27'].head(10)

0                 Healthcare and Social Assistance
1                                              NaN
2              Administrative and Support Services
3                                              NaN
4                 Public Administration/Government
5    Hospitality (Accommodation and Food Services)
6                                              NaN
7                  Professional and Other Services
8                     Education, Training and Arts
9                  Professional and Other Services
Name: Q27, dtype: object

In [ ]:
person_df['Q27'].value_counts(dropna=False)

NaN                                              703
Professional and Other Services                  212
Healthcare and Social Assistance                 126
Education, Training and Arts                      84
Construction                                      80
Retail Trade                                      74
Hospitality (Accommodation and Food Services)     72
Public Administration/Government                  68
Transportation & Logistics                        49
Manufacturing                                     43
Administrative and Support Services               19
Wholesale Trade                                    6
Utilities                                          6
Agriculture                                        3
Name: Q27, dtype: int64